sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -235,5 +258,17 @@ private ByteBufFlux bodyIntern() {
         Connection internConnection() {
             return reactorNettyConnection;
         }
+
+        private static ByteBuffer deepCopyBuffer(ByteBuf byteBuf) {
+            ByteBuffer buffer = byteBuf.nioBuffer();
+            int offset = buffer.position();
+            int size = buffer.remaining();
```

Comment:
minor; maybe not now, but we could have a static empty ByteBuffer in class level and if size == 0, we can just return that, instead of creating a new empty `duplicate` reference every time.